src/lib/assetBridger/l1l3Bridger.ts

```diff
@@ -0,0 +1,1382 @@
+import { Provider, TransactionRequest } from '@ethersproject/abstract-provider'
+import {
+  BigNumber,
+  BigNumberish,
+  Overrides,
+  PayableOverrides,
+  Signer,
+  ethers,
+} from 'ethers'
+import { IERC20 } from '../abi/IERC20'
+import { L2GatewayToken } from '../abi/L2GatewayToken'
+import { IL1Teleporter } from '../abi/IL1Teleporter'
+import { IERC20__factory } from '../abi/factories/IERC20__factory'
+import { L1GatewayRouter__factory } from '../abi/factories/L1GatewayRouter__factory'
+import { IL2ForwarderFactory__factory } from '../abi/factories/IL2ForwarderFactory__factory'
+import { L2GatewayToken__factory } from '../abi/factories/L2GatewayToken__factory'
+import { IL1Teleporter__factory } from '../abi/factories/IL1Teleporter__factory'
+import { Address } from '../dataEntities/address'
+import { ArbSdkError } from '../dataEntities/errors'
+import {
+  L1Network,
+  L2Network,
+  TeleporterAddresses,
+  l1Networks,
+  l2Networks,
+} from '../dataEntities/networks'
+import {
+  SignerOrProvider,
+  SignerProviderUtils,
+} from '../dataEntities/signerOrProvider'
+import { L1ToL2TransactionRequest } from '../dataEntities/transactionRequest'
+import {
+  L1ToL2MessageReader,
+  L1ToL2MessageStatus,
+} from '../message/L1ToL2Message'
+import { L1ToL2MessageCreator } from '../message/L1ToL2MessageCreator'
+import {
+  GasOverrides,
+  L1ToL2MessageGasEstimator,
+  PercentIncrease,
+} from '../message/L1ToL2MessageGasEstimator'
+import {
+  L1ContractCallTransaction,
+  L1ContractCallTransactionReceipt,
+  L1EthDepositTransactionReceipt,
+  L1TransactionReceipt,
+} from '../message/L1Transaction'
+import { Erc20Bridger } from './erc20Bridger'
+import { isDefined } from '../utils/lib'
+import { Inbox__factory } from '../abi/factories/Inbox__factory'
+import { OmitTyped } from '../utils/types'
+import { getAddress } from 'ethers/lib/utils'
+import { IL2Forwarder } from '../abi/IL2Forwarder'
+import { ERC20__factory } from '../abi/factories/ERC20__factory'
+
+type PickedTransactionRequest = Required<
+  Pick<TransactionRequest, 'to' | 'data' | 'value'>
+>
+
+export enum TeleportationType {
+  /**
+   * Teleporting to an ETH fee L3
+   */
+  Standard,
+  /**
+   * Teleporting the fee token to a custom fee L3
+   */
+  OnlyFeeToken,
+  /**
+   * Teleporting a non-fee token to a custom fee L3
+   */
+  NonFeeTokenToCustomFee,
+}
+
+export type TxRequestParams = {
+  txRequest: PickedTransactionRequest
+  l1Signer: Signer
+  overrides?: PayableOverrides
+}
+
+type RetryableGasValues = {
+  gasLimit: BigNumber
+  maxSubmissionFee: BigNumber
+}
+
+export type DepositRequestResult = {
```

Comment:
should types like this be renamed to something like `Erc20L1L3DepositRequestResult` to remove ambiguity?